cloudify/tests/test_builtin_workflows.py

```diff
@@ -660,6 +661,68 @@ def _get_node_instance(self, node_id):
         return self.env.storage.get_node_instances(node_id=node_id)[0]
 
 
+class TestCheckStatus(unittest.TestCase):
+    @workflow_test(path.join(
+        'resources',
+        'blueprints',
+        'test-check-status.yaml'))
+    def test_all_nodes_run(self, cfy_local):
+        self.assertRaises(
+            exceptions.WorkflowFailed,
+            cfy_local.execute, 'check_status'
+        )
+        pass_instances = cfy_local.storage.get_node_instances(
+            node_id='node_passing')
+        fail_instances = cfy_local.storage.get_node_instances(
+            node_id='node_failing')
+        noop_instances = cfy_local.storage.get_node_instances(
+            node_id='node_undefined')
+        assert len(pass_instances) == 1
+        assert len(fail_instances) == 1
+        assert len(noop_instances) == 1
+
+        pass_instance = pass_instances[0]
+        assert pass_instance['system_properties']
+        assert pass_instance['system_properties']['ok']
+
+        # the noop instance did run, even though it depends on a failing one
+        noop_instance = noop_instances[0]
+        assert noop_instance['system_properties']
+        assert noop_instance['system_properties']['ok']
```

Comment:
why should it be ok?

Reply:
I declare that instances who have no check defined at all, are considered green